shm/player.cpp

```diff
@@ -0,0 +1,49 @@
+#include "player.hpp"
+
+Player::Player(std::unique_ptr<Ship> &ship, int money, int availableSpace)
```

Comment:
Don't pass by reference, use `&&` or just without it. It is confusing when you pass it by reference and then move.